processor/transformprocessor/internal/metrics/func_agregate_label_metrics.go

```diff
@@ -0,0 +1,405 @@
+		return nil, fmt.Errorf("AggregateLabelFactory args must be of type *AggregateLabelArguments")
+	}
+
+	if !args.Type.IsValid() {
+		return nil, fmt.Errorf("AggregateLabel accepts one of the following functions: min/max/mean/sum")
+	}
+
+	return AggregateLabel(args.Type, args.LabelSet)
+}
+
+func AggregateLabel(aggregationType common.AggregationType, labelSetMap map[string]bool) (ottl.ExprFunc[ottlmetric.TransformContext], error) {
+	return func(_ context.Context, tCtx ottlmetric.TransformContext) (any, error) {
+		metric := tCtx.GetMetric()
+
+		filterAttrs(metric, labelSetMap)
+		newMetric := pmetric.NewMetric()
+		copyMetricDetails(metric, newMetric)
+		ag := groupDataPoints(metric)
+		mergeDataPoints(newMetric, aggregationType, ag)
+		newMetric.MoveTo(metric)
+
+		return nil, nil
+	}, nil
+}
+
+func copyMetricDetails(from, to pmetric.Metric) {
+	to.SetName(from.Name())
+	to.SetUnit(from.Unit())
+	to.SetDescription(from.Description())
+	//exhaustive:enforce
+	switch from.Type() {
+	case pmetric.MetricTypeGauge:
+		to.SetEmptyGauge()
+	case pmetric.MetricTypeSum:
+		to.SetEmptySum().SetAggregationTemporality(from.Sum().AggregationTemporality())
+		to.Sum().SetIsMonotonic(from.Sum().IsMonotonic())
+	case pmetric.MetricTypeHistogram:
+		to.SetEmptyHistogram().SetAggregationTemporality(from.Histogram().AggregationTemporality())
+	case pmetric.MetricTypeExponentialHistogram:
+		to.SetEmptyExponentialHistogram().SetAggregationTemporality(from.ExponentialHistogram().AggregationTemporality())
+	case pmetric.MetricTypeSummary:
+		to.SetEmptySummary()
+	}
+}
+
+func filterAttrs(metric pmetric.Metric, filterAttrKeys map[string]bool) {
+	if filterAttrKeys == nil {
+		return
+	}
+	rangeDataPointAttributes(metric, func(attrs pcommon.Map) bool {
+		attrs.RemoveIf(func(k string, _ pcommon.Value) bool {
+			return !filterAttrKeys[k]
+		})
+		return true
+	})
+}
+
+// rangeDataPointAttributes calls f sequentially on attributes of every metric data point.
+// The iteration terminates if f returns false.
+func rangeDataPointAttributes(metric pmetric.Metric, f func(pcommon.Map) bool) {
+	//exhaustive:enforce
+	switch metric.Type() {
+	case pmetric.MetricTypeGauge:
+		for i := 0; i < metric.Gauge().DataPoints().Len(); i++ {
+			dp := metric.Gauge().DataPoints().At(i)
+			if !f(dp.Attributes()) {
+				return
+			}
+		}
+	case pmetric.MetricTypeSum:
+		for i := 0; i < metric.Sum().DataPoints().Len(); i++ {
+			dp := metric.Sum().DataPoints().At(i)
+			if !f(dp.Attributes()) {
+				return
+			}
+		}
+	case pmetric.MetricTypeHistogram:
+		for i := 0; i < metric.Histogram().DataPoints().Len(); i++ {
+			dp := metric.Histogram().DataPoints().At(i)
+			if !f(dp.Attributes()) {
+				return
+			}
+		}
+	case pmetric.MetricTypeExponentialHistogram:
+		for i := 0; i < metric.ExponentialHistogram().DataPoints().Len(); i++ {
+			dp := metric.ExponentialHistogram().DataPoints().At(i)
+			if !f(dp.Attributes()) {
+				return
+			}
+		}
+	case pmetric.MetricTypeSummary:
```

Comment:
the summary type does no seem to be supported in the original aggregate function. I had a linting error when removing this case. That's why I let it here. But if you look closely, there is no aggregation logic behind this

Reply:
Thanks for the explanation. We should still have a test for summary types just so we test how they are handled.

If we can handle summary values and it doesn't take too long to implement, I think we should aggregate them here. If not, then we should return an error if a summary metric is passed to this function. We want users to know when they are providing invalid data, if they know they want summaries in the same pipeline where they do aggregations, they can add a check to not run this function when the metric is a summary.

Reply:
Ok will check what it takes to implement the summary, otherwise I will add the error when aggregating summary metrics

Reply:
Just for clarification, for aggregation of summary, only `Count` and `Sum` values should be considered for the given aggregation function? Or also data in `QuantileValues` parameter?

Thanks in advance!

Reply:
adapted to return error when using summary

Reply:
I'm fine leaving summaries out for now.

>Just for clarification, does it actually makes sense to create aggregation for Summaries? It's similar to ExponentialHistogram (if I understand it correctly).

To my knowledge, Summaries were originally only included for Prometheus compatibility. It would be nice to support any OTLP metric type that is possible to aggregate, Summaries included.